packages/api/lib/writeRecords/write-granules.js

```diff
@@ -429,6 +430,85 @@ const _writeGranule = async ({
   await publishGranuleSnsMessageByEventType(granuletoPublish, snsEventType);
 };
 
+/**
+ * Update granule record status to queued in DynamoDB, PostgreSQL, Elasticsearch.
+ * Publish SNS event for updated granule.
+ *
+ * @param {Object}  params
+ * @param {Object}  params.apiGranuleRecord      - API Granule object to write to
+ *                                                 the database
+ * @param {Object}  params.originalApiGranule    - API Granule object before status update
+ * @param {Object}  params.postgresGranuleRecord - PostgreSQL granule record to write
+ *                                                 to database
+ * @param {string}  params.granuleId             - Granule ID
+ * @param {Object}  params.granuleModel          - Instance of DynamoDB granule model
+ * @param {Object}  params.granulePgModel        - @cumulus/db compatible granule module instance
+ * @param {Knex}    params.knex                  - Knex object
+ * @param {string}  params.snsEventType          - SNS Event Type, defaults to 'Update'
+ * @param {Object}  params.esClient              - Elasticsearch client
+ * returns {Promise}
+ */
+const _writeQueuedGranule = async ({
+  apiGranuleRecord,
+  originalApiGranule,
+  postgresGranuleRecord,
+  granuleId,
+  granuleModel,
+  granulePgModel,
+  knex,
+  snsEventType = 'Update',
+  esClient,
+}) => {
+  const apiGranule = apiGranuleRecord;
+  delete apiGranule.execution;
+
+  let granuleCumulusId;
+  const status = 'queued';
```

Comment:
It seems like `status` on the `postgresGranuleRecord` and `apiGranuleRecord` comes into this function as `queued`. Really, this function is just a wrapper for doing a record "update" in all three database systems and publishing to SNS, which is a good thing. However, I don't think function needs to have its own logic for determining what status to set and could be more generic, like so:

```
const _updateGranule = async ({
  apiGranule,
  postgresGranule,
  fieldUpdates,
  dynamoFieldsToDelete,
  granuleModel,
  granulePgModel,
  knex,
  snsEventType = 'Update',
  esClient,
}) => {
```

where 

- `fieldUpdates` would be `{ status: 'queued'}`
- `dynamoFieldsToDelete` would be `['execution']`
- `apiGranule` would be the original granule from Dynamo
- `postgresGranule` would be the existing granule in PG

All of which would be passed in from `updateGranuleStatusToQueued`. It makes this function more generic and reduces overlap between it and `updateGranuleStatusToQueued`. 

`granuleId` can be destructured from `apiGranule`, so doesn't need to be passed in separately

Reply:
[`cfed85a` (#2593)](https://github.com/nasa/cumulus/pull/2593/commits/cfed85a44156433493bdb1170bc1a7075838fd1d)
What are your thoughts on this refactor? I have 
```
  const apiGranule = apiGranuleRecord;
  delete apiGranule.execution;
``` 
because trying to delete the execution field from `apiGranule` directly shows:
```
Assignment to property of function parameter 'apiGranuleRecord'.eslintno-param-reassign
```

Reply:
Yeah that makes sense. Let's just use lodash  like so:

`const esGranule = omit(apiGranule, dynamoFieldsToDelete)`

This way we also don't have to hardcode in deletion of just the `execution` field but can respect any fields being removed from the dynamo record. Also, I know it was my suggestion, but let's rename `dynamoFieldsToDelete` to `apiFieldsToDelete` since we'll use it for both Dynamo and Elasticsearch

Reply:
[`0f6d234` (#2593)](https://github.com/nasa/cumulus/pull/2593/commits/0f6d2348fd5d7b49057bc1ab688199dd9d66609e)

Reply:
[`21ea079` (#2593)](https://github.com/nasa/cumulus/pull/2593/commits/21ea0795e9f8eafaf839a99b7341acb262176ad5)